addon/serializers/json.js

```diff
@@ -1535,4 +1538,40 @@ if (isEnabled("ds-payload-type-hooks")) {
 
 }
 
+if (isEnabled("ds-serialize-id")) {
+
+  JSONSerializer.reopen({
+
+    /**
+     serializeId can be used to customize how id is serialized
+
+     For example, your server may expect integer datatype of id
+
+     ```app/serializers/application.js
+     import DS from 'ember-data';
+
+     export default DS.JSONSerializer.extend({
+     serializeId(snapshot, json, primaryKey) {
+       var id = snapshot.id;
+       json[primaryKey] = parseInt(id, 10);
+       }
+    });
+     ```
+
+     @method serializeId
+     @public
+     @param {DS.Snapshot} snapshot
+     @param {Object} json
+     @param {String} primaryKey
+     */
+    serializeId(snapshot, json, primaryKey) {
+      var id = snapshot.id;
+
+      if (id && primaryKey) {
```

Comment:
I think the check for `primaryKey` can be removed. This would then be symmetric to the implementation when this feature is not enabled.